src/main/java/com/google/api/codegen/transformer/PageStreamingTransformer.java

```diff
@@ -117,4 +118,43 @@ private PageStreamingDescriptorClassView generateDescriptorClass(
 
     return desc.build();
   }
+
+  public List<PageStreamingFactoryClassView> generateFactoryClasses(
+      SurfaceTransformerContext context) {
+    List<PageStreamingFactoryClassView> factories = new ArrayList<>();
+
+    context.getNamer().addPageStreamingFactoryImports(context.getTypeTable());
+    for (Method method : context.getPageStreamingMethods()) {
+      factories.add(generateFactoryClass(context.asMethodContext(method)));
+    }
+
+    return factories;
+  }
+
+  private PageStreamingFactoryClassView generateFactoryClass(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    ModelTypeTable typeTable = context.getTypeTable();
+    Method method = context.getMethod();
+    PageStreamingConfig pageStreaming = context.getMethodConfig().getPageStreaming();
+    FeatureConfig featureConfig = context.getFeatureConfig();
+
+    PageStreamingFactoryClassView.Builder fact = PageStreamingFactoryClassView.newBuilder();
+
+    Field resourceField = pageStreaming.getResourcesField();
+
+    fact.name(namer.getPageStreamingDescriptorConstName(method));
+    fact.requestTypeName(typeTable.getAndSaveNicknameFor(method.getInputType()));
+    fact.responseTypeName(typeTable.getAndSaveNicknameFor(method.getOutputType()));
+    fact.resourceTypeName(typeTable.getAndSaveNicknameForElementType(resourceField.getType()));
+    fact.pagedListResponseTypeName(
+        namer.getAndSavePagedResponseTypeName(
+            method,
+            featureConfig,
+            typeTable,
+            method.getInputType(),
```

Comment:
I think it is safe to assume these types are always the input and output types of the given method. Since you have already passed in the method, can we get rid of the input and output types here?